x-pack/legacy/plugins/integrations_manager/server/plugin.ts

```diff
@@ -17,11 +24,25 @@ export interface PluginInitializerContext {}
 
 export type PluginSetup = ReturnType<Plugin['setup']>;
 export type PluginStart = ReturnType<Plugin['start']>;
+export interface PluginContext {
+  esClient: ClusterClient;
+}
 
 export class Plugin {
   constructor(initializerContext: PluginInitializerContext) {}
   public setup(core: CoreSetup) {
-    const { server } = core.http;
+    const { http, elasticsearch } = core;
+    const { server } = http;
+
+    // make these items available to handlers via h.context
+    // https://github.com/hapijs/hapi/blob/master/API.md#server.bind()
+    // aligns closely with approach proposed in handler RFC
+    // https://github.com/epixa/kibana/blob/rfc-handlers/rfcs/text/0003_handler_interface.md
+    const context: PluginContext = {
+      esClient: elasticsearch.createClient(PLUGIN_ID),
+    };
+
+    server.bind(context);
```

Comment:
I can't think of a concrete example of how this might break something, but as far as I understand this will expose your context to all route handlers in all of kibana. This feels a bit dangerous and reaching beyond the scope of this plugin, especially if other plugins want to adopt this pattern.

Maybe you can wrap your route handlers in a context injector function? Something like (untested pseudo code):

```ts
const injectContext = (handler, context) => {
 return (req: Request) => handler(req, context)
}

routes
  .map(route => Object.assign({}, route, {handler: injectContext(route.handler, context)})
  .forEach(route => server.route(route));
```

Reply:
I see your point and the code sample you have (or something like it) works for me as well. I’ll update, add comments, and ping when it’s up. Thanks.

Reply:
Should be addressed by 7af0c921bf2369c88143271ccd49a0deb8e9b257. LMK if it’s not.